GVFS/GVFS.Hooks/Program.cs

```diff
@@ -452,5 +422,17 @@ private static string GetGitCommandSessionId()
                 return string.Empty;
             }
         }
+
+        private static bool IsUpgradeMessageDeterministic()
+        {
+            try
+            {
+                return Environment.GetEnvironmentVariable("GVFS_UPGRADE_DETERMINISTIC", EnvironmentVariableTarget.Process) != null;

```

Comment:
Just to check, getting an env variable is so cheap it wont affect performance?  Am I reading the code correctly that this will be run on every command?

Reply:
Environment variables are cheap. Reading Git config or a file on disk is problematic and slow.